### PR TITLE
[Improve][Doc] Add `file_filter_pattern` example to doc

### PR DESCRIPTION
**Pull request by YOMO-Lee**
Added file filtering instructions to the localfile connector documentation

### CODE REVIEW

File: docs/en/connector-v2/source/LocalFile.md [hunk 1]
```diff
@@ -254,6 +254,72 @@ Specifies Whether to process data using the tag attribute format.
 
 Filter pattern, which used for filtering files.
 
+The filtering format is similar to wildcard matching file names in Linux.
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/LocalFile.md]: We cannot tell users about the ambiguous conclusion. Please tell users directly that we use Java regular expressions.

File: docs/en/connector-v2/source/LocalFile.md [hunk 2]
```diff
@@ -254,6 +254,72 @@ Specifies Whether to process data using the tag attribute format.
...
+| [!]          | Match any single character not in parentheses                                                                                  | [!abc]* &emsp; Any file that does not start with abc                                                                                         |
+| [a-z]        | Match any single character from a to z                                                                                         | [a-z]* &emsp; Any file starting with a to z                                                                                                  |
+| {a,b,c}/a..z | When separated by commas, it represents individual characters<br/>When separated by two dots, represents continuous characters | {a,b,c}* &emsp; Files starting with any character from abc<br/>{a..Z}* &emsp;&ensp; Files starting with any character from a to z            |
+
+However, it should be noted that unlike Linux wildcard characters, when encountering file suffixes, the middle dot cannot be omitted.
+
+For example, `abc20241022.csv`, the normal Linux wildcard `abc*` is sufficient, but here we need to use `abc*.*` , Pay attention to a point in the middle.
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/LocalFile.md]: Please replace this part to link to https://en.wikipedia.org/wiki/Regular_expression. Let user to learn regular itself.

File: docs/en/connector-v2/source/LocalFile.md [hunk 3]
```diff
@@ -254,6 +254,72 @@ Specifies Whether to process data using the tag attribute format.
...
+abcw20241022.csv
+abcx20241022.csv
+abcq20241022.csv
+abcg20241022.csv
+abcv20241022.csv
+abcb20241022.csv
+old_data.csv
+logo.png
+script.sh
+helpers.sh
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/LocalFile.md]: Please add some file path, not only match file name.

**Review by Hisoka-X — COMMENTED** [review]
Thanks @YOMO-Lee ! I left some comments.

File: docs/en/connector-v2/source/LocalFile.md [hunk 4]
```diff
@@ -254,6 +254,54 @@ Specifies Whether to process data using the tag attribute format.
 
 Filter pattern, which used for filtering files.
 
+The pattern follows standard regular expressions. For details, please refer to https://en.wikipedia.org/wiki/Regular_expression. learn it
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/LocalFile.md]:
```suggestion
The pattern follows standard regular expressions. For details, please refer to https://en.wikipedia.org/wiki/Regular_expression.
```

File: docs/en/connector-v2/source/LocalFile.md [hunk 5]
```diff
@@ -254,6 +254,54 @@ Specifies Whether to process data using the tag attribute format.
...
+
+File Structure Example:
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/LocalFile.md:260]:
```suggestion
There are some examples.

File Structure Example:
```

**Comment by YOMO-Lee** [comment]
@Hisoka-X Please review this

**Review by Hisoka-X — APPROVED** [review]

**Comment by YOMO-Lee** [comment]
@zhilinli123 please review

**Review by zhilinli123 — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**